src/loggers/database-logger.php

```diff
@@ -61,10 +61,13 @@ public function logger( $query, $time ) {
 	 * @return void
 	 */
 	public function log_output() {
+		$content_type = $this->get_content_type();
+
 		if (
 			\wp_doing_ajax()  ||
 			( defined( 'WP_CLI' ) && WP_CLI ) ||
-			( defined( 'REST_REQUEST' ) && REST_REQUEST )
+			( defined( 'REST_REQUEST' ) && REST_REQUEST ) ||
+			( $content_type !== 'text/html' && $content_type !== '' )
```

Comment:
The content-type header may contain other information, like the charset.
I'd like to suggest the following:
```php
		$content_type          = $this->get_content_type();
		$non_html_content_type = stripos( $content_type, 'text/html' ) > - 1
		                         && $content_type !== '';
		if (
			\wp_doing_ajax()
			|| ( defined( 'WP_CLI' ) && WP_CLI )
			|| ( defined( 'REST_REQUEST' ) && REST_REQUEST )
			|| $non_html_content_type
		) {
			return;
		}
```

Reply:
Basically if `stripos` returns `false`, and `$content_type` is not `''`, right? would rather have the logic in the `if`.